ql/instruments/makecms.cpp

```diff
@@ -201,8 +201,8 @@ namespace QuantLib {
 
     MakeCms& MakeCms::withDiscountingTermStructure(
                 const Handle<YieldTermStructure>& discountingTermStructure) {
-        engine_ = boost::shared_ptr<PricingEngine>(new
-                            DiscountingSwapEngine(discountingTermStructure));
+        engine_ = boost::static_pointer_cast<PricingEngine>(
```

Comment:
Ok, is it just to be 110% safe or did you actually see a compiler having problems with it? It would be interesting to know.